lib/plugins/stackoverflow/post.dart

```diff
@@ -32,11 +35,86 @@ class StackOverflowPostOwner with _$StackOverflowPostOwner {
       _$StackOverflowPostOwnerFromJson(json);
 }
 
-String postToDiscordMarkdown(StackOverflowPost post) {
-  return '# ${post.title}\n'
+Future<String> postToDiscordMarkdown(
+  StackOverflowPost post,
+  Environment env,
+) async {
+  String header = '# ${post.title}\n'
       '[${post.owner.displayName}](<${post.owner.link}>) - '
       '${timeago.format(DateTime.fromMillisecondsSinceEpoch(post.creationDate * 1000))} | '
       '${post.score} votes | '
-      '[Source](<${post.link}>)\n\n'
-      '${post.bodyMarkdown}';
+      '[Source](<${post.link}>)\n\n';
+
+  String bodyMarkdown = env.pastebinApiKey == null
+      ? post.bodyMarkdown
+      : await shortenWithPastebin(
+          post.bodyMarkdown,
+          PastebinClient(env.pastebinApiKey!),
+          kMaxMessageLength - header.length,
+        );
+
+  String result = '$header$bodyMarkdown';
+
+  if (result.length > kMaxMessageLength) {
+    return '${result.substring(0, kMaxMessageLength - 3)}...';
+  }
+
+  return result;
+}
+
+// Matches code blocks fenced with ```code``` in group 1 and code blocks
+// indented by four spaces in group 2 (SO seems to use the latter quite often).
```

Comment:
doesnt the API we use from SO guarantee we get correct markdown, 
so there should be no such thing as 4-space indented code?
did you actually come accross such a thing during testing?

Reply:
Four space indented code is valid markdown - and yes, I came across it several times during testing. I'm wondering if SO's rich text editor uses four spaces if you use the UI to create a code block (as opposed to writing it yourself).

Reply:
you learn something new every day. incredible.